### ISSUE TITLE
need help with maven deployment

### ISSUE THREAD

**Issue by codejanovic** [yegor256/rultor#1120]
hey guys,

im still playing around with your great rultor service to deploy a java project to maven. now that everything seems to build, the maven deployment is not executed by rultor and i was wondering why.

i would be very thankful if someone may take a look at my project configuration and the latest buildlog: 

* project https://github.com/codejanovic/guice-autowire
* buildlog http://www.rultor.com/t/8896-233265982

thanks in advance,
aleks

**Comment by alex-palevsky** [comment]
@original-brownbear please pay attention to this issue ([par.21](http://at.teamed.io/policy.html#21))

**Comment by original-brownbear** [comment]
@codejanovic hey sorry for the delay.

Looks to me like the `yml` in your `.rultor.yml` isn't valid so Rultor uses the default, you're missing a `-` ;)
See https://github.com/yegor256/rultor/blob/master/.rultor.yml#L24 for any example.

**Comment by codejanovic** [comment]
@original-brownbear theres really no need to apologize. 
i doublechecked the `.rultor.yml` and compared it with an existing project called `ezyfox-sfs2x` (https://github.com/youngmonkeys/ezyfox-sfs2x/blob/master/.rultor.yml)
But as of the logs of `ezyfox-sfs2x` (http://www.rultor.com/t/9022-232231444) the `.rultor.yml` could be read by Rultor. 

I will add the missing dash and head back to you ASAP. Thanks for your time!

**Comment by codejanovic** [comment]
@original-brownbear i added the missing dash in `rultor.yml` without success. any other ideas?

as rultor behaves like everything is okay i have no chance to check whats wrong. if he is falling back to default behaviour because of an invalid `rultor.yml` he should at least log (big and fat) that the given `rultor.yml` isn't valid (even if he is not able to tell why exactly).

**Comment by original-brownbear** [comment]
@codejanovic I agree about the throwing on invalid `yml`, we do have an issue for that open :)

As for your specific case, I think the problem is that you fixed the `yml` in `develop`, but Rultor then builds from master, again using the broken one :(

**Comment by codejanovic** [comment]
@original-brownbear okay thats nice, looking forward to this feature!

actually i simply forgot to merge the PR from develop -> master, so i did it now with the same result i guess :-( http://www.rultor.com/t/8896-236042255

**Comment by original-brownbear** [comment]
@codejanovic looks like we could have a bug here, couldn't find anything obviously wrong with your `.rultor.yml` now. I mean if you want and got the time, could you try adding a `merge:` and `deploy:` section also to see if their absence is causing trouble with Rultor's parser ?

**Comment by codejanovic** [comment]
@original-brownbear oh my .... just realised after looking for your `qulice` repo that i named the file `rultor.yml` instead of `.rultor.yml` - i guess thats the reason why rultor falls back to its defaults -.- gonna rename and retry it

**Comment by codejanovic** [comment]
Yep, thats looks much better now, even though the build is faling http://www.rultor.com/t/8896-236459544

**Comment by original-brownbear** [comment]
@codejanovic looks like it passed to me ? :)

**Comment by codejanovic** [comment]
@original-brownbear well, i also wasnt able to find an error in the log, but `rultor` says he failed https://github.com/codejanovic/guice-autowire/issues/1

**Comment by original-brownbear** [comment]
@codejanovic ah it's this issue #1016 :) you didn't actually have any commit to merge.

**Comment by codejanovic** [comment]
@original-brownbear so whats the desired workflow on this? triggering `rultor` from within an open PR?

**Comment by original-brownbear** [comment]
@codejanovic ah no ... my bad, it's not that bug. You're apparently running an empty commit as your last command. This appears to have exit code not equal to `0` making the build fail right ?

**Comment by codejanovic** [comment]
@original-brownbear well yes, i only changed the file name. last try before going to sleep 👍

**Comment by codejanovic** [comment]
@original-brownbear so i know im late to the party but i tried it again with slight modifications to `.rultor.yml` and it did not work either. 
i really dont see any hints in the log though :-( current log: http://www.rultor.com/t/8896-248373269

**Comment by codejanovic** [comment]
@original-brownbear i managed to get one step further. mvn deploy is now getting triggered by rultor, but fails with ([full log](http://www.rultor.com/t/8896-265012377))

```
[ERROR] Rule failure while trying to close staging repository with ID "iogithubcodejanovic-1004".
[ERROR] 
[ERROR] Nexus Staging Rules Failure Report
[ERROR] ==================================
[ERROR] 
[ERROR] Repository "iogithubcodejanovic-1004" failures
[ERROR]   Rule "signature-staging" failures
[ERROR]     * Missing Signature: '/io/github/codejanovic/guice-autowire/0.1.0/guice-autowire-0.1.0-sources.jar.asc' does not exist for 'guice-autowire-0.1.0-sources.jar'.
[ERROR]     * Missing Signature: '/io/github/codejanovic/guice-autowire/0.1.0/guice-autowire-0.1.0.jar.asc' does not exist for 'guice-autowire-0.1.0.jar'.
[ERROR]     * Missing Signature: '/io/github/codejanovic/guice-autowire/0.1.0/guice-autowire-0.1.0.pom.asc' does not exist for 'guice-autowire-0.1.0.pom'.
[ERROR]     * Missing Signature: '/io/github/codejanovic/guice-autowire/0.1.0/guice-autowire-0.1.0-javadoc.jar.asc' does not exist for 'guice-autowire-0.1.0-javadoc.jar'.
[ERROR] 
[ERROR] 
[ERROR] Cleaning up local stage directory after a Rule failure during close of staging repositories: [iogithubcodejanovic-1004]
[ERROR]  * Deleting context 20b73625799caa.properties
[ERROR] Cleaning up remote stage repositories after a Rule failure during close of staging repositories: [iogithubcodejanovic-1004]
[ERROR]  * Dropping failed staging repository with ID "iogithubcodejanovic-1004" (Rule failure during close of staging repositories: [iogithubcodejanovic-1004]).
```

i think the problem maybe lies in settings.xml and the provided profile id, as the log also says

```
[WARNING] The requested profile "guice-autowire" could not be activated because it does not exist.
```

now heres my `settings.xml` (without the real credentials) and i am not sure what exactly to put in as profile id. i tried both, *guice-autowire* and *codejanovic/guice-autowire* but none of them work

```xml
<settings>
  <profiles>
    <profile>
      <id>codejanovic/guice-autowire</id>
      <properties>
        <gpg.homedir>/home/r</gpg.homedir>
        <gpg.keyname>mykey id</gpg.keyname>
        <gpg.passphrase>mykey pass</gpg.passphrase>
      </properties>
    </profile>
  </profiles>
  <servers>
    <server>
      <id>oss.sonatype.org</id>
      <username>myusername</username>
      <password>mypw
    </server>
  </servers>
</settings>
```

any thoughts on this?

**Comment by yegor256** [comment]
@codejanovic is this still an issue? If not, please close this ticket.

**Comment by edmoffo** [comment]
Closing as invalid: per https://github.com/yegor256/rultor/issues/1120#issuecomment-236459464 the original config file was named `rultor.yml` rather than `.rultor.yml`, which is why Rultor fell back to its defaults; subsequent failures in the thread were GPG and Sonatype settings issues outside Rultor's contract. Open a new issue if a similar problem appears against the current code.
